test/unit/Cardano/EnvironmentSpec.hs

```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Cardano.EnvironmentSpec where
+
+import Prelude
+
+import Cardano.Environment
+    ( ErrMissingOrInvalidEnvVar (..), Network, unsafeLookupEnv )
+import Data.Maybe
+    ( isNothing )
+import Data.Proxy
+    ( Proxy (..) )
+import Data.Text.Class
+    ( TextDecodingError (..) )
+import System.Environment
+    ( setEnv, unsetEnv )
+import Test.Hspec
+    ( Spec, describe, it, shouldThrow )
+import Test.QuickCheck
+    ( Arbitrary (..) )
+import Test.QuickCheck.Arbitrary.Generic
+    ( genericArbitrary, genericShrink )
+import Test.Text.Roundtrip
+    ( textRoundtrip )
+
+spec :: Spec
+spec = do
+    describe "Can perform roundtrip textual encoding & decoding" $ do
+        textRoundtrip $ Proxy @Network
+
+    describe "ErrMissingOrInvalidEnvVar (Show / displayException)" $ do
+        let errNoAdditionalContext = ErrMissingOrInvalidEnvVar
+                { name = "PATATE"
+                , command = "my-command"
+                , additionalContext = Nothing
+                }
+        let errWithAdditionalContext = ErrMissingOrInvalidEnvVar
+                { name = "PATATE"
+                , command = "my-command"
+                , additionalContext = Just
+                    ("💩"
+                    , TextDecodingError
+                        { getTextDecodingError = "not a valid value" }
+                    )
+                }
+        it (show errNoAdditionalContext) True
+        it (show errWithAdditionalContext) True
```

Comment:
What's the purpose of those two _tests_?

Reply:
The `displayException` and `show` method are quite verbose on this type (to display rich error message in the console when something goes wrong). So the test here are really about _showing_ what the output looks like so that we can control it, at least from a visual perspective.